tools/gfx/debug-layer.cpp

```diff
@@ -706,12 +724,24 @@ void DebugRenderCommandEncoder::setStencilReference(uint32_t referenceValue)
     return baseObject->setStencilReference(referenceValue);
 }
 
+void DebugRenderCommandEncoder::writeTimestamp(IQueryPool* pool, SlangInt index)
+{
+    SLANG_GFX_API_FUNC;
+    baseObject->writeTimestamp(static_cast<DebugQueryPool*>(pool)->baseObject, index);
```

Comment:
It's possible to do something like a dynamic cast for incoming com interfaces by giving the impls their own guid, and querying for that, This can be seen implemented in CacheFileSystem::queryInterface(SlangUUID const& uuid, void** outObject). By adding `SLANG_CLASS_GUID(0x2f4d1d03, 0xa0d1, 0x434b, { 0x87, 0x7a, 0x65, 0x5, 0xa4, 0xa0, 0x9a, 0x3b })` and then doing 

```
 if (uuid == CacheFileSystem::getTypeGuid())
    {
        *outObject = this;
        return SLANG_OK;
    }
```

Note does not increment reference count. 

This some what outside of 'COM' but is 'safe' under the assumption that Guids don't clash (they are unique!), and that the GUID you are using is private to your impl. 

This is not *necessary* but thought I'd throw it out there.